services/base.js

```diff
@@ -241,6 +262,15 @@ class BaseService {
       })
     )
   }
+
+  static logDebug(...args) {
```

Comment:
Should we pass a log level here so we can make better use of logging levels depending on the context (`console.info()`, `console.error()`, `console.warn()` etc) here instead of just using `console.log()` for everything?

That would also allow `debug()` to return a log level instead of true/false to help filter events if we wanted.